qutebrowser/mainwindow/tabwidget.py

```diff
@@ -439,11 +445,24 @@ def tabSizeHint(self, index):
             # get scroll buttons as soon as needed.
             size = minimum_size
         else:
+            #TODO: relative size and/or configured one
+            try:
+                tab = objreg.get('tab', scope='tab', window=self._win_id, tab=index)
+                if tab.pin is True:
+                    size = QSize(40, height)
+                    qtutils.ensure_valid(size)
+                    return size
```

Comment:
This block should be in an `else:` after the `except`, so the exception wouldn't be caught if it happens here for some reason (see [this part](https://docs.python.org/3/tutorial/errors.html#handling-exceptions) of the python docs)